test/fixtures/process/gc-not-close.mjs

```diff
@@ -0,0 +1,21 @@
+import { strictEqual } from 'assert'
+
+function setup() {
+  let obj = { foo: 'bar' }
+  process.finalization.register(obj, shutdown)
+  setImmediate(function () {
+    obj = undefined
+    gc()
```

Comment:
I removed and the test failed, so I will keep it.

Reply:
Does it fail because of `strictEqual(shutdownCalled, false)` below? Why does it expect the shutdown callback to never be called? That sounds like a flake risk because GC is not deterministic. If it counts on GC behavior to pass, the next time V8 updates its GC strategy this can start flaking in the CI.